racket/collects/racket/private/immediate-default.rkt

```diff
@@ -0,0 +1,35 @@
+(module kw '#%kernel
+  (#%require "define.rkt"
+             "small-scheme.rkt"
+             "stxcase-scheme.rkt"
+             (for-template '#%kernel))
+
+  (#%provide immediate-default?)
+
+  ;; A default-argument expression counts as an "immediate default"
```

Comment:
I did not measure how often a default-argument expression is an immediate default. Informally, it happens "a lot". Not the metric you're asking for, but (since it occurs to me that I forgot to rebuild the bootstrapped expander) the expander's expansion in pretty-printed form goes from 80k lines to 78.2k lines.

Variables can't be treated as immediate defaults, because they might be mutable; treating them as immediate might reorder evaluations. Looking for immutable variables is complicated at best and doesn't seem worthwhile.

Reply:
Ah, that makes sense. I might try adding the instrumentation and counting; if so I'll report back here.